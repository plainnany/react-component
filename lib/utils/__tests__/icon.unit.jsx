import renderer from 'react-test-renderer'
import React from 'react'
import { mount } from 'enzyme'
import Icon from '../../icon/icon'

describe('icon', () => {
  it('是个icon render successfully', () => {
    const json = renderer.create(<Icon name="wechat" />).toJSON()
    expect(json).toMatchSnapshot()
  })

  it('触发点击事件', () => {
    const fn = jest.fn()
    const component = mount(<Icon name="wechat" onClick={fn} />) 
    component.find('svg').simulate('click')
    expect(fn).toBeCalled()
  })
})