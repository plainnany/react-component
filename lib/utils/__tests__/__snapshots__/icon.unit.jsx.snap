// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`icon 是个icon render successfully 1`] = `
<span>
  <svg
    className="icon"
  >
    <use
      xlinkHref="#wechat"
    />
  </svg>
</span>
`;
